server/server_test.go

```diff
@@ -213,6 +213,50 @@ func TestGetConnectURLs(t *testing.T) {
 	}
 }
 
+func TestClusterClientAdvertiseConnectURL(t *testing.T) {
+	opts := DefaultOptions()
+	opts.Port = 4222
+	opts.Cluster.ClientAdvertiseStr = "nats.example.com"
+
+	s := New(opts)
+
+	urls := s.getClientConnectURLs()
+
+	if len(urls) != 1 {
+		t.Fatalf("Expected to get one url, got none: %v with Cluster.AdvertiseStr %v",
+			opts.Host, opts.Cluster.ClientAdvertiseStr)
+	}
+
+	if urls[0] != "nats.example.com:4222" {
+		t.Fatalf("Expected to get '%s', got: '%v'", "nats.example.com:4222", urls[0])
+	}
+	s.Shutdown()
+
+	opts.Cluster.ClientAdvertiseStr = "nats.example.com, nats2.example.com:7777"
```

Comment:
You could also try not specifying port for more than one address, or at least not just the first to check that port is given to all addresses that miss it, not just the first?